elooplib/threadpool.hpp

```diff
@@ -4,86 +4,42 @@
 #include <memory>
 #include <utility>
 #include <thread>
+#include <chrono>
 #include <mutex>
 #include <atomic>
 #include <future>
 #include <condition_variable>
 #include <queue>
 
-class ThreadPool
+class ITask
 {
 public:
-    ThreadPool(size_t num_threads)
-    {
-        threads.reserve(num_threads);
-        for (size_t i = 0; i < num_threads; i++)
-        {
-            threads.emplace_back(&ThreadPool::run, this);
-        }
-    }
-    ThreadPool(ThreadPool&) = delete; 
-    ThreadPool& operator=(ThreadPool&) = delete; 
-    ThreadPool(ThreadPool&&) = delete;
-    ThreadPool& operator=(ThreadPool&&) = delete; 
-                                   //threads(std::move(rhs.threads)){};
+    virtual void run(){};
+    ITask(){};
 
-    template <typename Func, typename... Args>
-    void add_task(const Func &task_func, Args &&...args)
-    {
-        std::cout << "Add obj to threadpool\n";
-        std::lock_guard<std::mutex> q_lock(q_mtx);
-        t_queue.emplace(std::async(std::launch::deferred, task_func, args...));
-        q_cv.notify_one();
-    }
+    virtual ~ITask(){};
```

Comment:
`virtual ~ITask() = default`

or
`virtual ~ITask();`
and in cpp file:
`ITask::~ITask() = default`